### PR TITLE
Fireman carry qol from /tg/

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports https://github.com/tgstation/tgstation/pull/49733 https://github.com/tgstation/tgstation/pull/51539
You can place people you're carrying onto tables or smash their head on harm intent
You can place them into morgue/cremator trays and automatically place and buckle them to stasis units and other buckleable objects

## Why It's Good For The Game
quality of life

## Changelog
:cl: lordScrubling, Mothblocks, Rohesie
add: placing fireman carried people onto things
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Idk about the throwing, considering how powerful throwing is and being able to run at full speed with someone on your back just to throw at someone in combat

**Comment by Cenrus** [comment]
Remove the throwing, it is too good and body throws should be a situational, tricky but rewarding thing to pull off.

**Comment by BrumalLake** [comment]
not full speed, not that powerful, still situational
you can't fireman carry someone in the middle of a fight

**Comment by PowerfulBacon** [comment]
If you know an antagonist is in a place, you can pickup a corpse run in and throw it for a pretty long stun that cannot be blocked, easilly long enough to stun and take them out / at least steal their weapons

**Comment by BrumalLake** [comment]
it's a less than a second knockdown, and you can't shovestun
you can do the same thing by dragging a body

**Comment by Cenrus** [comment]
Wrong, it is a two second paralyze. Especially important since all knockdowns outside of disarms don't make you drop things.

**Comment by BrumalLake** [comment]
Well you can still do the same thing by dragging someone with you

**Comment by BrumalLake** [comment]
> 
> 
> If you know an antagonist is in a place, you can pickup a corpse run in and throw it for a pretty long stun that cannot be blocked, easilly long enough to stun and take them out / at least steal their weapons

2 seconds isnt easily enough time, especially with their body covering whatever they dropped

**Comment by Cenrus** [comment]
That's the point, drag slowdown exists, while fireman carrying lets you run at (almost) max speed. this is not a 'QOL' PR.

**Comment by BrumalLake** [comment]
yeah, ignore placing people on stasis units on stuff
this pr is purely for combat
dont people complain about sec being too strong on a daily basis
i dont know whos gonna be hanging around a dead body all the time
its not like people throw people in every fight
its already difficult to hit someone moving
i will nerf collisions if thats what people want but throwing is useful and i wont get rid of it

**Comment by BrumalLake** [comment]
kept it as a paralyze cause i didnt want people shovestunning for a 4 second paralysis instead

**Comment by PowerfulBacon** [comment]
we want fireman carried bodies to not be throwable, not nerfed body impacts

**Comment by BrumalLake** [comment]
I thought the problem was the stun? Whats wrong with throwing them other than the impact?

**Comment by Archanial** [comment]
>qol
>balance/rebalance

make up your mind

**Comment by BrumalLake** [comment]
> 
> 
> > qol
> > balance/rebalance
> 
> make up your mind

its just the title you guys have no chill
its intended to be quality of life, look at the tg pr damnn

**Comment by BrumalLake** [comment]
happy now?

**Comment by DatBoiTim** [comment]
If you're gonna change fireman carries how bout you fix them first?

**Comment by BrumalLake** [comment]
> 
> 
> If you're gonna change fireman carries how bout you fix them first?

https://github.com/BeeStation/BeeStation-Hornet/pull/4666

**Comment by DatBoiTim** [comment]
> > 
> 
> > 
> 
> > If you're gonna change fireman carries how bout you fix them first?
> 
> 
> 
> https://github.com/BeeStation/BeeStation-Hornet/pull/4666

Excellent

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -190,15 +190,43 @@
...
+					tableplace_delay = 2.75 SECONDS
+					skills_space = " quickly"
+				carried_mob.visible_message("<span class='notice'>[user] begins to[skills_space] place [carried_mob] onto [src]...</span>",
+					"<span class='userdanger'>[user] begins to[skills_space] place [carried_mob] onto [src]...</span>")
+				if(do_after(user, tableplace_delay, target = carried_mob))
+					user.unbuckle_mob(carried_mob)
+					tableplace(user, carried_mob)
+			else
+				user.unbuckle_mob(carried_mob)
+				tablepush(user, carried_mob)
```
Comment (PowerfulBacon) [inline code/game/objects/structures/tables_racks.dm:218]: Is grab and disarm intent meant to tdo the exact same thing as help intent but instantly? Also does this count as table slamming?

Comment (BrumalLake) [reply code/game/objects/structures/tables_racks.dm:218]: yeah grab and disarm are tableslams so the rider should get up after if they are not already resting

**Review by qwertyquerty — APPROVED** [review]

**Pull request merged**